File: client-clojure/src/sirqul_io_t_platform/specs/account_list_response.clj
(ns sirqul-io-t-platform.specs.account-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.account-response :refer :all]
            )
  (:import (java.io File)))


(def account-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of account-response-spec)
   (ds/opt :count) int?
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :start) int?
   (ds/opt :limit) int?
   (ds/opt :countTotal) int?
   (ds/opt :returning) string?
   })

(def account-list-response-spec
  (ds/spec
    {:name ::account-list-response
     :spec account-list-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/AccountListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AccountListResponse._

case class AccountListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[AccountResponse]],
count: Option[Integer],
hasMoreResults: Option[Boolean],
start: Option[Integer],
limit: Option[Integer],
countTotal: Option[Long],
returning: Option[String])

object AccountListResponse {
  import DateTimeCodecs._

  implicit val AccountListResponseCodecJson: CodecJson[AccountListResponse] = CodecJson.derive[AccountListResponse]
  implicit val AccountListResponseDecoder: EntityDecoder[AccountListResponse] = jsonOf[AccountListResponse]
  implicit val AccountListResponseEncoder: EntityEncoder[AccountListResponse] = jsonEncoderOf[AccountListResponse]
}
